TLM/TLM/Util/Extensions/NetLaneExtensions.cs

```diff
@@ -9,5 +9,15 @@ public static class NetLaneExtensions {
             ((NetLane.Flags)netLane.m_flags).CheckFlags(
                 required: NetLane.Flags.Created,
                 forbidden: NetLane.Flags.Deleted);
+
+        /// <summary>
+        /// Checks if the netLane is Created, but not Deleted and if its netSegment is Created, but neither Collapsed nor Deleted.
+        /// </summary>
+        /// <param name="netLane">netLane</param>
+        /// <returns>True if the netLane and its netSegment is valid, otherwise false.</returns>
+        public static bool IsValidWithSegment(this ref NetLane netLane) {
```

Comment:
Some mods like MOM add a buggy lane without segment. This might confuse TMPE. but I am not sure if that extra lane can even have traffic so probably not relevant to TMPE. in any case you haven't changed the logic of the code so I don't expect regression.

Reply:
I hope I didnt change any logic. but better double check during review in case I mistyped or something. :)